src/panels/lovelace/cards/hui-entity-button-card.ts

```diff
@@ -217,15 +222,18 @@ class HuiEntityButtonCard extends LitElement implements LovelaceCard {
   }
 
   private _computeBrightness(stateObj: HassEntity | LightEntity): string {
-    if (!stateObj.attributes.brightness) {
+    if (
+      !stateObj.attributes.brightness ||
+      this._config?.state_color === false
+    ) {
       return "";
     }
     const brightness = stateObj.attributes.brightness;
     return `brightness(${(brightness + 245) / 5}%)`;
   }
 
   private _computeColor(stateObj: HassEntity | LightEntity): string {
-    if (!stateObj.attributes.hs_color) {
+    if (!stateObj.attributes.hs_color || this._config?.state_color === false) {
```

Comment:
Is there a reason that this can't be `!this._config?.state_color`? Since you are defaulting it to true?

Reply:
It can be `undefined`